config/config.go

```diff
@@ -295,6 +295,16 @@ func (cfg *Config) addTransports(h host.Host) error {
 //
 // This function consumes the config. Do not reuse it (really!).
 func (cfg *Config) NewNode() (host.Host, error) {
+	// If possible check that the resource manager conn limit is higer than the
+	// limit set in the conn manager.
+	l, ok := cfg.ResourceManager.(connmgr.GetterConnLimit)
+	if ok {
+		err := cfg.ConnManager.CheckLimit(l)
+		if err != nil {
+			log.Warn(err)
```

Comment:
It'd be nice to augment the warning with the fact that it comes from the resource manager:
something like this:
`rcmgr limit conflicts with connmgr limit: conn manager high water limit: 8, exceeds the system limit of: 4`